src/source.ts

```diff
@@ -32,13 +33,30 @@ export default class Source extends React.Component<Props, void> {
 
     if (!map.getSource(this.id) && (geoJsonSource || tileJsonSource)) {
       map.addSource(this.id, (geoJsonSource || tileJsonSource) as any);
+      map.on('load', this.onData);
 
       if (onSourceAdded) {
         onSourceAdded(map.getSource(this.id) as GeoJSONSource | TilesJson);
       }
     }
   }
 
+  private onData = (event: any) => {
+    const { map } = this.context;
+
+    const source = map.getSource(this.props.id) as GeoJSONSource;
+
+    const { onSourceLoaded } = this.props;
+    if (onSourceLoaded) {
+      onSourceLoaded(source);
+    }
+    // Will fix datasource being empty
+    if (this.props.geoJsonSource && this.props.geoJsonSource.data) {
```

Comment:
Can you please add a check for source here, as the event could be fired by another component that add a source to the map, which could eventually result to trying to access `setData` of `undefined`